src/app/_components/product.tsx

```diff
@@ -0,0 +1,221 @@
+"use client";
+import { Switch } from "../_components/ui/switch";
+import React, { useState } from "react";
+import { Button, buttonVariants } from "../_components/ui/button";
+import Image from "next/image";
+
+interface Product {
+  id: number;
+  name: string;
+  description: string | null;
+  calification: string | null;
+  personalize: string[];
+  price: number;
+  restaurant: string | null;
+  imageUrl?: string | null;
+}
+
+interface ProductProps {
+  product: Product;
+  variant?: "default" | "mobile";
+}
+
+const Product = ({ product, variant = "default" }: ProductProps) => {
+  const { name, description, calification, personalize, price, restaurant } =
+    product || {};
```

Comment:
I think you don't need `|| {}` here since product is always defined. Apart from that, I personally don't always destructuring, it's the most standard thing and we usually do it when we have just a few props, but when you have to destructure from an object with 15 props and you do it this way, that is definitely too much. In my opinion this amount of properties is still fine (specially since prettier didn't moved them to different lines), but it's touching the limit of being too much. No action needed on this last part of the comment, but I think it's something to keep in mind.